base/iobuffer.jl

```diff
@@ -92,6 +92,13 @@ end
     return byte
 end
 
+@inline function unsafe_read(from::AbstractIOBuffer, ::Type{UInt8})
```

Comment:
maybe we should mark the tests in `read` as `@boundscheck`. but this method is an abuse of what the `unsafe_` prefix should mean.